src/sentry/static/sentry/app/components/events/relatedEvents/index.tsx

```diff
@@ -0,0 +1,184 @@
+import moment from 'moment-timezone';
+import {Location} from 'history';
+
+import space from 'app/styles/space';
+import {t} from 'app/locale';
+import {Organization, Event} from 'app/types';
+import EventView from 'app/utils/discover/eventView';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import {getTraceDateTimeRange} from 'app/components/events/interfaces/spans/utils';
+import EventDataSection from 'app/components/events/eventDataSection';
+import {PanelTable, PanelItem} from 'app/components/panels';
+import PlatformIcon from 'app/components/platformIcon';
+import Link from 'app/components/links/link';
+import Button from 'app/components/button';
+import DiscoverQuery from 'app/utils/discover/discoverQuery';
+import DateTime from 'app/components/dateTime';
+import {IconOpen} from 'app/icons';
+import TableView from 'app/views/eventsV2/table/tableView';
+import SortLink from 'app/components/gridEditable/sortLink';
+
+import TypeIndicator from './typeIndicator';
+
+type Type = React.ComponentProps<typeof TypeIndicator>['type'];
+
+type Props = {
+  organization: Organization;
+  event: Event;
+  location: Location;
+};
+
+type state = {
+  traceEvents: Array<TraceEvent>;
+  isLoading: boolean;
+};
+
+type TraceEvent = {
+  'event.type': string;
+  id: string;
+  project: string;
+  'project.id': number;
+  timestamp: string;
+  title: string;
+  'trace.span': string;
+};
+
+// List events that have the same tracing ID as the current Event
+class RelatedEvents extends React.Component<Props, state> {
+  state: state = {traceEvents: [], isLoading: true};
+
+  componentDidMount() {
+    //this.getIssues();
+  }
+
+  // async getIssues() {}
+
+  getEventView = () => {
+    const {organization, event} = this.props;
+    const traceId = event.contexts?.trace?.trace_id;
+
+    const orgFeatures = new Set(organization.features);
+    const dateCreated = moment(event.dateCreated).valueOf() / 1000;
+    const pointInTime = event?.dateReceived
+      ? moment(event.dateReceived).valueOf() / 1000
+      : dateCreated;
+
+    const {start, end} = getTraceDateTimeRange({start: pointInTime, end: pointInTime});
+
+    return EventView.fromSavedQuery({
+      id: undefined,
+      name: `Events with Trace ID ${traceId}`,
+      fields: ['title', 'event.type', 'project', 'trace.span', 'timestamp'],
+      orderby: '-timestamp',
+      query: `trace:${traceId}`,
+      projects: orgFeatures.has('global-views') ? [] : [Number(event.projectID)],
+      version: 2,
+      start,
+      end,
+    });
+  };
+
+  render() {
+    const {organization, location} = this.props;
+    const orgSlug = organization.slug;
+
+    const eventView = this.getEventView();
+
+    return (
+      <EventDataSection
+        type="relted-events"
+        title={t('Related Events')}
+        actions={<Button size="small">{t('Open in Discover')}</Button>}
+      >
+        <DiscoverQuery location={location} eventView={eventView} orgSlug={orgSlug}>
```

Comment:
Not all organizations have access to discover. You'll need a `<Feature>` gate around this.